subsys/bluetooth/audio/tmap.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/types.h>
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <stdlib.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include "audio_internal.h"
+#include <zephyr/bluetooth/audio/tmap.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bt_tmap, CONFIG_BT_TMAP_LOG_LEVEL);
+
+static uint16_t tmap_role;
+static struct bt_tmap_cb *cb;
+static bool tmas_found;
+
+static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
+static struct bt_gatt_discover_params discover_params;
+static struct bt_gatt_read_params read_params;
+
+uint8_t tmap_char_read(struct bt_conn *conn, uint8_t err,
+					struct bt_gatt_read_params *params,
+					const void *data, uint16_t length)
+{
+	enum bt_tmap_role peer_role = sys_get_le16(data);
+
+	if (cb->tmap_discovery_complete) {
+		cb->tmap_discovery_complete(peer_role, conn, err);
+	}
+	return BT_GATT_ITER_STOP;
+}
+
+static uint8_t discover_func(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+							struct bt_gatt_discover_params *params)
+{
+	int err;
+
+	if (!attr) {
+		(void)memset(params, 0, sizeof(*params));
+		if (tmas_found == false) {
+			/* TMAS not found on peer */
+			if (cb->tmap_discovery_complete) {
+				cb->tmap_discovery_complete(0, conn, -EINVAL);
```

Comment:
Suggest to either use `BT_ATT_ERR_ATTRIBUTE_NOT_FOUND` or `EOPNOTSUPP` instead of `EINVAL`